api/src/main/java/com/redhat/ipaas/api/v1/rest/exception/IPaasServerException.java

```diff
@@ -0,0 +1,28 @@
+package com.redhat.ipaas.api.v1.rest.exception;
+
+public class IPaasServerException extends RuntimeException {
```

Comment:
I think that what's missing here is a method that take a throwable and either wrap it inside an `IPaasServerException` or re-throw it if it a runtimeexception (e.g. IllegalArgumentException or EntityNotFoundException).